### PR TITLE
Add API route to handle Adyen 3DS redirection callback

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Create new API route.

Add page prop to payment page to show success state.

Add query param to payment page to communicate success 3DS redirection.

Add Adyen return URL to the configuration.

## Justify why they are needed

Just moving pieces from web-onboarding setup.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #471** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/471" target="_blank">Graphite</a> 
    * **PR #476** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/476" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/476?utm_source=stack-comment).

File: apps/store/src/pages/api/payment/adyen-callback/[locale].ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { AuthStatus } from '@/components/CheckoutPaymentPage/CheckoutPaymentPage.constants'
+import { PageLink } from '@/lib/PageLink'
+import { isLocale } from '@/utils/isLocale'
+
+export const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { MD: md, PaRes: pares } = req.body
+
+  if (!isLocale(req.query.locale)) return res.status(400).json({ message: 'Invalid locale' })
```
Comment (simonauner) [inline apps/store/src/pages/api/payment/adyen-callback/[locale].ts:9]: Should perhaps use `logger` to also log to Datadog?

File: apps/store/src/lib/PageLink.ts [hunk 2]
```diff
@@ -14,11 +16,17 @@ export const PageLink = {
   cart: ({ locale }: BaseParams = {}) => `${getOptionalPath(locale)}/cart`,
   cartReview: ({ locale }: BaseParams = {}) => `${getOptionalPath(locale)}/cart/review`,
   checkout: ({ locale }: BaseParams = {}) => `${getOptionalPath(locale)}/checkout`,
-  checkoutPayment: ({ locale }: BaseParams = {}) => `${getOptionalPath(locale)}/checkout/payment`,
+  checkoutPayment: ({ locale, authStatus }: CheckoutPaymentPage = {}) => {
+    const authStatusQueryParam = authStatus ? `authStatus=${authStatus}` : null
+    const queryString = authStatusQueryParam ? `?${authStatusQueryParam}` : ''
```
[the remark this answers is not in the document]
Comment (robinandeer) [reply apps/store/src/lib/PageLink.ts:21]: Haha yeah it's not very pretty - we could use `URLSearchParams` but need to polyfill it in node :P

File: apps/store/src/pages/api/payment/adyen-callback/[locale].ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/pages/api/payment/adyen-callback/[locale].ts:9]: Yes, for sure! Haven't really got there with coverage store yet but perhaps we should start thinking about it already :)

File: apps/store/src/lib/PageLink.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alebedev) [reply apps/store/src/lib/PageLink.ts:21]: How about https://github.com/sindresorhus/query-string ?

It's small and provider higher-level API

**Review by alebedev — APPROVED** [review]

File: apps/store/src/lib/PageLink.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (robinandeer) [reply apps/store/src/lib/PageLink.ts:21]: Yup sure, the second use case we find -- let's add it ;)

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/476?utm_source=stack-comment).


Job Reference: D7PCCahe3PVdqM8XBcQY

**Pull request merged**